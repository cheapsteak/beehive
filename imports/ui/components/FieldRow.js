// Framework
import React from "react";

// Components
import { Container, Row, Col, FormGroup, Label, Input } from "reactstrap";

// Helpers
import { camelCaseToTitle } from "../helpers/text";

// Types

/**
 * A single input field for a boolean, string or number.
 *
 * @param {Object} props All the properties passed to the React Component.
 * @returns {Element} Stateless functional React component.
 */
const FieldRow = ({
  label = "Field",
  value,
  visibleLabel = true,
  stateSettingAction
}) => {
  const inputId = `${label}-${typeof value}-field-row-input`;
  const titleLabel = camelCaseToTitle(label);
  const labelClass = visibleLabel ? "" : "sr-only";
  const onChange = ({ target: { value } }) => stateSettingAction(label, value);
  return (
    <FormGroup className="field-row">
      <Label className={labelClass} for={inputId}>
        {titleLabel}
      </Label>
      <Input
        type="text"
        placeholder={titleLabel}
        name={inputId}
        id={inputId}
        onChange={onChange}
        value={value}
      />
    </FormGroup>
  );
};

export default FieldRow;
