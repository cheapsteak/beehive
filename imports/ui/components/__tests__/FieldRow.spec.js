import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import FieldRow from '../FieldRow';
import { Input, Label } from 'reactstrap';

describe('<FieldRow />', () => {
  let defaultProps;

  describe('FieldRow SnapShot', () => {
    beforeEach(() => {
      defaultProps = {
        label: 'NewField',
        value: 15,
        visibleLabel: false,
        stateSettingAction: jest.fn()
      };
    });

    it('renders FieldRow correctly with default props', () => {
      delete defaultProps.label;
      delete defaultProps.visibleLabel;
      const tree = ReactTestRenderer.create(
        <FieldRow {...defaultProps} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders FieldRow correctly', () => {
      const tree = ReactTestRenderer.create(
        <FieldRow {...defaultProps} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders FieldRow correctly', () => {
      const tree = ReactTestRenderer.create(
        <FieldRow {...defaultProps} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('FieldRow Props', () => {
    let wrapper, input, label;

    beforeEach(() => {
      const wrapper = shallow(<FieldRow {...defaultProps} />);
      input = wrapper.find(Input);
    });

    it('should pass fields prop to FieldRows', () => {
      input.simulate('change', { target: { value: 'My new value' } });
      expect(defaultProps.stateSettingAction).toHaveBeenCalled();
    });
  });
});
