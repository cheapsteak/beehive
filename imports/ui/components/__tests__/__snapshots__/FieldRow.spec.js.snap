// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldRow /> FieldRow SnapShot renders FieldRow correctly 1`] = `
<div
  className="field-row form-group"
>
  <label
    className="sr-only "
    htmlFor="NewField-number-field-row-input"
  >
    New Field
  </label>
  <input
    className="form-control"
    id="NewField-number-field-row-input"
    name="NewField-number-field-row-input"
    onChange={[Function]}
    placeholder="New Field"
    type="text"
    value={15}
  />
</div>
`;

exports[`<FieldRow /> FieldRow SnapShot renders FieldRow correctly 2`] = `
<div
  className="field-row form-group"
>
  <label
    className="sr-only "
    htmlFor="NewField-number-field-row-input"
  >
    New Field
  </label>
  <input
    className="form-control"
    id="NewField-number-field-row-input"
    name="NewField-number-field-row-input"
    onChange={[Function]}
    placeholder="New Field"
    type="text"
    value={15}
  />
</div>
`;

exports[`<FieldRow /> FieldRow SnapShot renders FieldRow correctly with default props 1`] = `
<div
  className="field-row form-group"
>
  <label
    className=""
    htmlFor="Field-number-field-row-input"
  >
    Field
  </label>
  <input
    className="form-control"
    id="Field-number-field-row-input"
    name="Field-number-field-row-input"
    onChange={[Function]}
    placeholder="Field"
    type="text"
    value={15}
  />
</div>
`;
